resource-managers/kubernetes/core/src/test/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorLifecycleTestUtils.scala

```diff
@@ -71,6 +71,27 @@ object ExecutorLifecycleTestUtils {
       .build()
   }
 
+  def runningExecutorWithFailedContainer(executorId: Long): Pod = {
```

Comment:
Maybe add a comment on the purpose of this function? Namely setting the pod to failed iff the executor container fails. Also might this make sense to be configurable? I can imagine people who have sidecars for stuff like istio, if the sidecar fails the pod should fail since the networking isn't going to work. WDYT?